### PR TITLE
Juniper SRX

### PR DESCRIPTION
**Pull request by yossifirouz**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/JuniperSRX/ModelingRules/JuniperSRX/JuniperSRXModelingRules.xif [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+| alter tag_name = arrayindex(regextract(_raw_log ,"\:\s(\w+\_\w+)"),0)
+
+| alter XDM.Audit.event_timestamp = timestamp 
+| alter XDM.Audit.operation= hostname
+| alter XDM.Audit.identity.name= process
+| alter XDM.Audit.operation_type= tag_name
+| alter XDM.Audit.identity.uuid= processID
+| alter XDM.Audit.TriggeredBy.identity.name= username
+| alter XDM.Audit.threat.description= message
+| alter XDM.Audit.operation_type= tag_name
```
Comment (evisochek) [inline Packs/JuniperSRX/ModelingRules/JuniperSRX/JuniperSRXModelingRules.xif:21]: missing ; in the end

File: Packs/JuniperSRX/ModelingRules/JuniperSRX/JuniperSRXModelingRules.xif [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+| alter XDM.Network.Destination.ipv4= Destination_Address 
+| alter XDM.Network.Destination.port= Nat_Destination_Port 
+| alter XDM.Network.rule= Source_Nat_Rule_Name 
+| alter XDM.Network.ip_protocol= Protocol_ID 
+| alter XDM.Network.rule= Policy_Name 
+| alter XDM.Network.Source.zone= Source_Zone
+| alter  XDM.Network.Destination.zone= Destination_Zone_Name
+| alter XDM.Network.Destination.user.username= Username 
+| alter XDM.Network.Source.user.employee_id= Roles 
+| alter XDM.Network.Destination.packets= Packet_Incoming_Interface
```
Comment (evisochek) [inline Packs/JuniperSRX/ModelingRules/JuniperSRX/JuniperSRXModelingRules.xif:80]: missing ; in the end

File: Packs/JuniperSRX/ModelingRules/JuniperSRX/JuniperSRXModelingRules.xif [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+| alter XDM.Audit.identity.uuid= processID
+| alter XDM.Audit.TriggeredBy.identity.name= username
+| alter XDM.Audit.threat.description= message
+| alter XDM.Audit.operation_type= tag_name
+
+
+
+[MODEL: dataset="Juniper_SRX_raw", model="Network"] 
+
+| filter raw_log contains "source-address="
```
Comment (evisochek) [inline Packs/JuniperSRX/ModelingRules/JuniperSRX/JuniperSRXModelingRules.xif:27]: remove the pipe "|"

File: Packs/JuniperSRX/ModelingRules/JuniperSRX/JuniperSRXModelingRules.xif [hunk 4]
```diff
@@ -0,0 +1,83 @@
+[MODEL: dataset="Juniper_SRX_raw", model="Audit"] 
+
+| alter event_type = arrayindex(regextract(_raw_log, "(\w+\[\d+\]\:\s)"),0)
```
Comment (evisochek) [inline Packs/JuniperSRX/ModelingRules/JuniperSRX/JuniperSRXModelingRules.xif:3]: remove the pipe "|"

File: Packs/JuniperSRX/ModelingRules/JuniperSRXModelingRules.xif [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+| alter Nat_Source_Address = arrayindex(regextract(raw_log ,"nat\-source\-address\=\"([^\"]+)") ,0)
+| alter Service_Name = arrayindex(regextract(raw_log ,"service\-name=\"([^\"]+)") ,0)
+| alter Nat_Destination_IP = arrayindex(regextract(raw_log ,"nat\-destination\-address\=\"(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})") ,0)
+// extract source_ip
+| alter source_address_ipv4 = arrayindex(regextract(raw_log ,"source\-address\=\"(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})"), 0)
+| alter source_adderss_ipv6 =  arrayindex(regextract(raw_log ,"source\-address\=\"(\w+\:\w+\:\w+\:\w+\:\w+\:\w+\:\w+\:\w+)"), 0)
+| alter sourceip = coalesce(source_address_ipv4 ,source_adderss_ipv6)
+// end extract source_ip
+| alter XDM.Network.Source.ipv4= SourceIP
+| alter XDM.Network.event_timestamp = parse_timestamp("%FT%R%E*S", Timestamp)
```
Comment (guytamir10) [inline Packs/JuniperSRX/ModelingRules/JuniperSRXModelingRules.xif:56]: need to change the parse string to:

%FT%H:%M:%E3S

**Review by evisochek — APPROVED** [review]

**Pull request merged**